src/tequilapi-client.ts

```diff
@@ -227,22 +231,26 @@ export class HttpTequilapiClient implements TequilapiClient {
     await this.http.put(`identities/${id}/email`, { email })
   }
 
-  public async authChangePassword(
-    username: string,
-    oldPassword: string,
-    newPassword: string
-  ): Promise<void> {
-    return this.http.put(`auth/password`, {
-      username,
-      // eslint-disable-next-line @typescript-eslint/camelcase
-      oldPassword: oldPassword,
-      // eslint-disable-next-line @typescript-eslint/camelcase
-      newPassword: newPassword,
+  public authSetToken(token: string): void {
+    this.http.setHeaders({
+      Authorization: 'Bearer ' + token,
```

Comment:
Since this is a http client wrapper, it should be able to automatically set auth token, and handle it. I suppose for the forst version is fine to do this manually but there should be an option.

Reply:
Then using token issuing endpoint `authAuthenticate()` caller decides on his custom auth flow & token persistence.
Still improved contract with `authAuthenticate(request: AuthRequest, useToken: true)`